requestrouter.cpp

```diff
@@ -238,6 +236,9 @@ namespace libtremotesf::impl {
         if (!mSessionId.isEmpty()) {
             request.setRawHeader(sessionIdHeader, mSessionId);
         }
+        if (mConfiguration.authentication) {
+            request.setRawHeader(QByteArray("Authorization"), QByteArray(mBasicAuthHeaderValue));
```

Comment:
Please make header name a constant, like sessionIdHeader.

Reply:
Do you also want a constant for the `Basic ` authentication scheme prefix?

Reply:
The only thing `QByteArrayLiteral` would do here is remove one extra allocation in `setConfiguration` when creating `QByteArray("Basic")`. New QByteArray will still be immediately created on heap when calling `append` (also this happens only when server configuration is loaded, not for every request). It's still nice, but I think it would be simpler to just lift the whole header value into constant, as `QStringLiteral("Basic %1:%2")`, and the initialize `mBasicAuthHeaderValue` from it.

BTW copying resulting `mBasicAuthHeaderValue` for each request is kinda cheap because `QByteArray` is copy-on-write i.e. non-modified copies share the same underlying buffer (deep copy would still happen somewhere down the line when Qt writes headers to socket).

Reply:
> [...]  I think it would be simpler to just lift the whole header value into constant, as QStringLiteral("Basic %1:%2"), and the initialize mBasicAuthHeaderValue from it.

I'm not sure how that should work though: the value of the header isn't `Basic ${username}:${password}`, it's `Basic ${base64(normalize(concat(username, ":", password)))}`. Perhaps you meant `"Basic %1"`?

Something like this?

```cpp
const auto authorizationHeaderValueFormat = QString("Basic %1");
// snip...
mAuthorizationHeaderValue = authorizationHeaderValueFormat.arg(base64Credentials).toUtf8();
```

I'd be happy to do something like that, if you prefer.

Reply:
Opps, you are right. I think it looks good now.